code/modules/mob/living/simple_animal/simple_animal.dm

```diff
@@ -334,6 +336,11 @@
 		for(var/i in loot)
 			new i(loc)
 
+/mob/living/simple_animal/revive()
+	..()
+	if(initial_density)

```

Comment:
This is band-aid over the problem. Somewhere in the code, someone messed up setting the density. This means that if the density change for whatever legit reason (For example, a future potion), it will always get set back to its original value on revive.

This just introduces another bug for the future

Reply:
Would changing initial_density to living_density and setting it in death() be better? Death() is where the density is set to 0 and it isn't reset on revive.

Reply:
Looked at what tg did.

They just set density = initial(density) in revive.

The approach is OK but not the method.

initial is a proc (function) that returns the initial, compile time value of something. There's no need to have initial_density set in initialize